MetaMask/metamask-extension
# Refactor eth_getEncryptionPublicKey handling

## Explanation
Moves `encryption-public-key-manager` and related `metamask-controller` logic, responsible for `eth_getEncryptionPublicKey`, to a new `EncryptionPublicKeyController` that uses `EncryptionPublicKeyManager` from `@metamask/message-managers`.

See: https://github.com/MetaMask/MetaMask-planning/issues/394

## Manual Testing Steps

- Go to https://metamask.github.io/test-dapp/
- Connect MM wallet
- Go to the Encrypt / Decrypt section
- Click on "Get Encryption Key"
- If request is rejected, "Encryption key" field should display "Error: MetaMask EncryptionPublicKey: User denied message EncryptionPublicKey."
- If request is approved, the encryption key should appear.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
